src/twin/substrate.rs

```diff
@@ -39,10 +39,15 @@ where
         Ok(client)
     }
 
-    pub async fn update_twin(&self, kp: &KeyPair, relay: Option<String>) -> Result<()> {
+    pub async fn update_twin(&self, twin_id: u32, kp: &KeyPair, relay: Option<String>) -> Result<()> {
```

Comment:
The function is called `update_twin` hence it only update the twin with the given identity and whatever information passed to the the function.

Validation if the twin exists is not the responsibility of this function. 

Update is only called by the peer on boot, if the twin does not exist it will never reach this anyway